client/app.jsx

```diff
@@ -1,8 +1,27 @@
-import React from 'react';
-import Home from './pages/home';
+import React, { useState, useEffect } from 'react';
+import Header from './components/header';
+import PageContainer from './components/page-container';
+import Auth from './pages/auth-page';
+import parseRoute from './lib/parse-route';
 
-export default class App extends React.Component {
-  render() {
-    return <Home />;
-  }
+function App() {
+  // eslint-disable-next-line no-unused-vars
+  const [route, setRoute] = useState(parseRoute(window.location.hash));
+
+  useEffect(() => {
+    window.addEventListener('hashchange', () => {
+      setRoute(parseRoute(window.location.hash));
+    });
+  });
```

Comment:
Here you want to add a second argument to `useEffect` of an empty array so it will act like `componentDidMount` and only get called one time instead of every time it re-renders.
```JAVASCRIPT
useEffect(() => {
  window.addEventListener('hashchange', () => {
    setRoute(parseRoute(window.location.hash));
  });
}, []);
```